views/default/page/elements/comments.php

```diff
@@ -38,30 +38,60 @@
 	'container_guid' => $entity->guid,
 	'full_view' => true,
 	'limit' => $limit,
+	'offset' => (int) get_input('offset'),
 	'distinct' => false,
 	'url_fragment' => $module_vars['id'],
 	'order_by' => [new OrderByClause('e.guid', $latest_first ? 'DESC' : 'ASC')],
 	'list_class' => 'comments-list',
+	'pagination' => true,
+	'preload_owners' => true,
 ];
 
-$show_guid = (int) elgg_extract('show_guid', $vars);
-if ($show_guid && $limit) {
-	// show the offset that includes the comment
-	// this won't work with threaded comments, but core doesn't support that yet
-	$operator = $latest_first ? '>' : '<';
-	$condition = function(QueryBuilder $qb) use ($show_guid, $operator) {
-		return $qb->compare('e.guid', $operator, $show_guid, ELGG_VALUE_INTEGER);
-	};
-	$count = elgg_count_entities([
-		'type' => 'object',
-		'subtype' => 'comment',
-		'container_guid' => $entity->guid,
-		'wheres' => [$condition],
-	]);
-	$options['offset'] = (int) floor($count / $limit) * $limit;
+$module_title = false;
+
+if (!$entity instanceof \ElggComment) {
+	$module_title = elgg_echo('comments');
+	
+	$options['metadata_name_value_pairs'] = ['level' => 1];
+	
+	$show_guid = (int) elgg_extract('show_guid', $vars);
+	if ($show_guid && $limit) {
+		// show the offset that includes the comment
+		$operator = $latest_first ? '>' : '<';
+		$condition = function(QueryBuilder $qb) use ($show_guid, $operator) {
+			return $qb->compare('e.guid', $operator, $show_guid, ELGG_VALUE_INTEGER);
+		};
```

Comment:
if this is only used once just add it to the `wheres` directly

Reply:
👍